### PR TITLE
Answer to Exercise 5-7 of Chapter 6

### PR DESCRIPTION
**Pull request by ssmylh**
@axel22 Please review.

### CODE REVIEW

File: src/main/scala/org/learningconcurrency/exercises/ch6/Ex6.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def update(k: K, v: V): Unit = map.get(k) match {
+      case Some(s) => s.onNext(v)
+      case _ =>
+        val s = Subject[V]()
+        map(k) = s
+        s.onNext(v)
+    }
+
+    /* This method throws `NoSuchElementException` if the key does not exist in the map. */
+    def apply(k: K): Observable[V] = map.get(k).get
```
Comment (axel22) [inline src/main/scala/org/learningconcurrency/exercises/ch6/Ex6.scala]: It should not throw an exception if there is no such element.

Instead, if there is a subscription, the reactive map should create an observable X such that when somebody subscribes to X, a key with the subject is added to the map if it does not already exist, and the observable subscribes to that subject. If the subscription is canceled (`unsubscribe`), the subject is removed from the map, unless somebody stored an actual value to that key.

This way, map does not grow indefinitely (no memory leaks), and the semantics do not require that there is actually a key in the map - i.e. one component can subscribes to the key, and wait until the key appears.

**Comment by axel22** [comment]
Thanks for the PR! Please see my comment about `RMap`.

**Comment by ssmylh** [comment]
Thanks for you comment !
I will fix `RMap`.

**Comment by ssmylh** [comment]
@axel22 I fixed `RMap`.
Please review.

**Comment by axel22** [comment]
LGTM

Thanks a lot!

**Comment by ssmylh** [comment]
I have learned a lot from you, thank you :smile:

**Pull request merged**